core/trino-main/src/main/java/io/trino/execution/SqlStage.java

```diff
@@ -263,7 +262,7 @@ public void recordGetSplitTime(long start)
         stateMachine.recordGetSplitTime(start);
     }
 
-    private synchronized void updateTaskStatus(TaskStatus status)
+    private void updateTaskStatus(TaskStatus status)
```

Comment:
it looks like with this change you can stay in `RUNNING` state even if all tasks so far finished executing.

Reply:
Why do you think so?